beacon-chain/blockchain/chain_info.go

```diff
@@ -68,19 +70,43 @@ func (s *Service) HeadRoot() []byte {
 
 // HeadBlock returns the head block of the chain.
 func (s *Service) HeadBlock() *ethpb.BeaconBlock {
+	if s.headBlock == nil {
+		head, err := s.beaconDB.HeadBlock(s.ctx)
```

Comment:
Not sure how I feel about this. This feels reckless. I think im going to change it to, if `s.headblock == nil` then return genesis block